.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,26 @@
+name: Checkstyle
+
+on: [push]
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+
+      - name: Run Checkstyle
+        run: |
+          mvn checkstyle:check
+          checkstyleExitCode=$?
+          if [ $checkstyleExitCode -ne 0 ]; then
+            echo "Checkstyle found issues. Please fix them before committing."
+            exit 1
+          fi
```

Comment:
is this part needed? I thought the above command would have failed the build if it found an issue

Reply:
Confirmed that build succeeds without lines 22-26. removed.